### ISSUE TITLE
Deep Space Satellites Plugin!!

### ISSUE THREAD

**Issue by SpaceDodger** [Stellarium/stellarium#259]
Hi Hello I'm Using Stellarium For Past 4 Years, I Yesterday Updated 0.18.1 To 0.18.2 Version That is very Nice  🙂, But I wish you an Request Can you Do a New Plugin Called Deep Space Satellites Plugin for Stellarium So that must be Awful to Use More and more , I Hope you can Do that Just Linking NASA's Eyes Data to your Software , Even though NASA's Eyes Software Uses Stellarium Constellations Only, So Please Try to Provide a New Plugin on Upcoming Updates, 

Your's Faithful of Space Lover
Pramodh Braj @SpaceDodger

**Comment by axd1967** [comment]
could this be comparable to comet orbits?

**Comment by gzotti** [comment]
Not really, except when you get compatible orbital elements and times when they should switch. Interplanetary spacecraft usually need swingby maneuvres to gain or lose speed and change course to final target, and then are deviated during flyby or captured into orbit at their target object. Occasionally, also rocket propulsion is used to change the Kepler-like orbit. I have visualized the original Rosetta trajectory (to comet 46P/Wirtanen) once (in a different project) that was available in a data file from ESA. Here I visualized the "instantaneous Kepler orbit", which changed at every encounter.  It had a few encounters where the orbit "switched". Very fascinating, and certainly on the long list of "want to have it also here". But other things come first, at least from my side.

**Comment by axd1967** [comment]
Here is another input: compute deep space probe flares.

> Apparently, on Aug. 26,  the UAE mission "Al-Amal" was detected at .078 AU by the Catalina Sky Survey and initially (mis?)catalogued as C332UJ2.

This means that it _might_ be possible to detect deep space probes near the antisolar point (ASP) or in some geometrical configuration related to that ASP, on the condition that they use large solar panels.

So, assuming the presence of large solar panels, the plugin might be able to compute the reflection of sunlight towards Earth, and thus indicate whether a probe _might_ be visible (compare this to Iridium flares).

Literally a shot in the dark, but might be interesting info for larger telescopes.

https://twitter.com/planet4589/status/1298992343240937472

**Comment by axd1967** [comment]
The plugin could also use the JPL Horizons Ephemeris service to retrieve probe positions, but it doesn't look very API friendly (telnet??) So an option might be to require the user to manually invoke Horizons, and then copy-paste this in a text file that is read by the plugin.

# workaround

1. Use the Horizon web interface to retrieve RA/DEC (ICRF 2000)
(https://ssd.jpl.nasa.gov/?horizons)
1. use equatorial J2000 on the Position tab of the Search window (ctrl-F) to center the display
1. put a Marker (shift + mouse) on the center of FOV marker

example for Hope mission on Aug 26, 000UTC:

![image](https://user-images.githubusercontent.com/3529789/92302145-10769480-ef6a-11ea-965c-04c7452962f0.png)

```
*******************************************************************************************************************************************************************************************************
 Date__(UT)__HR:MN     R.A._____(ICRF)_____DEC   APmag   S-brt     Illu%            delta      deldot    S-O-T /r    S-T-O    O-P-T     PlAng Cnst L_Ap_Hour_Ang      phi  PAB-LON  PAB-LAT App_Lon_Sun
*******************************************************************************************************************************************************************************************************

 2020-Aug-26 00:00     00 47 04.41 +26 01 06.7    n.a.    n.a.  83.51486 0.07640990589873   3.3144966 128.8709 /L  47.9133     n.a. -17.07846  And -02 11 58.452  47.9100 357.8821  11.1472        n.a.
```

RA: 00 47 04.41
DEC: +26 01 06.7

![image](https://user-images.githubusercontent.com/3529789/92302282-4d8f5680-ef6b-11ea-9890-a2c6a1b7c9bb.png)
